Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -395,8 +397,8 @@ public static LookupElementBuilder createLookupElement(@NotNull final Project pr
       }
 
       // If this is a class, try to show which package it's coming from.
-      if (element.getKind().equals(ElementKind.CLASS) && suggestion.getElementUri() != null) {
-        lookup = lookup.appendTailText(" (" + suggestion.getElementUri() + ")", true);
+      if (element.getKind().equals(ElementKind.CLASS) && displayUri != null) {
```

Comment:
BTW, this means that when the user completes not a class, e.g. a typedef, or a function, we don't show the URI. It seems sad.

Reply:
Yes, good point. Looking through https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html#type_ElementKind do you think there are other ElementKinds that we could include here that would make sense without changing the UI? We discussed offline, but there's an issue in the case of a function that the parameters (already in parentheses) could conflict with the package info.

Reply:
For everything that is not a function, we can show the URI.

And for functions probably display parameters.

Reply:
Sounds good, done!

Reply:
@scheglov, @lambdabaa, should we also skip showing URI for types like `ElementKind.SETTER`, `ElementKind.METHOD`, `ElementKind.CONSTRUCTOR`, `ElementKind.CONSTRUCTOR_INVOCATION`, or any other `ElementKind`?

Reply:
Good point. Maybe the right thing to do then is

```
if (displayUri != null &&
    element.getTypeParameters() == null &&
    element.getParameters() == null) {
  lookup = lookup.appendTailText(" (" + displayUri + ")", true);
}
```

since that's exactly when there won't be conflicts. Wdyt?

Reply:
Sounds reasonable. Done as you suggest. Happy to hear @scheglov's comments.

Reply:
Thank you, I think this is as good as we can do keeping the same approach. So, we can see that `json` top-level variant is from `dart:convert`. We cannot see that `jsonDecode` is also from `dart:convert`, but there is no space for it :-(

![image](https://user-images.githubusercontent.com/384794/53698476-12b6cc00-3d92-11e9-9d9b-96d616445698.png)